frontends/p4/structInitializers.cpp

```diff
@@ -28,26 +28,129 @@ convert(const IR::Expression* expression, const IR::Type* type) {
         auto si = new IR::IndexedVector<IR::NamedExpression>();
         if (auto le = expression->to<IR::ListExpression>()) {
             size_t index = 0;
+            auto l = le->size();
             for (auto f : st->fields) {
+                if (index >= l) {
+                    break;
+                }
                 auto expr = le->components.at(index);
                 auto conv = convert(expr, f->type);
                 auto ne = new IR::NamedExpression(conv->srcInfo, f->name, conv);
                 si->push_back(ne);
                 index++;
             }
+            if (le->defaultInitializer) {
+                modified = true;
+                IR::Expression* expr;
+                for (; index < st->fields.size(); index++) {
+                    auto f = st->fields.at(index);
+                    if (f->type->is<IR::Type_Bits>()) {
```

Comment:
I would make a separate function for this, `IR::Expression* defaultValue(const IR::Type*)`.